lib/plugins/inventory.js

```diff
@@ -427,13 +427,24 @@ function inject (bot, { version, hideErrors }) {
   function confirmTransaction (windowId, actionId, accepted) {
     // drop the queue entries for all the clicks that the server did not send
     // transaction packets for.
-    let click = windowClickQueue.shift()
-    if (click === undefined) {
+    // Also reject transactions that aren't sent from mineflayer
+    let click = windowClickQueue[0]
```

Comment:
Can you explain the logic?
This function is "confirmTransaction"
What your new code is doing is whenever we want to confirm a transaction
And there is no click in the queue we first reject the transaction
Why is that ?

Reply:
We reject the transaction if the specific transaction id is not in windowClickQueue, or if there is no transaction ids in windowClickQueue.

If we don't, the server can send an out-of-order transaction packet, that will mess up how we respond.
For example, say we click a slot in the inventory. The server may send an different response to that click and then the valid response. 

Example for Mineflayer:
Mineflayer sends click window with action id of 1
Server sends transaction with action id of -1064
---- (Vanilla sends transaction with action id of -1064)
Server sends transaction with action id of 1
Mineflayer sends transaction with action id of 1

Reply:
Code from Vanilla:
public void handleContainerAck(ClientboundContainerAckPacket clientboundContainerAckPacket) {
  PacketUtils.ensureRunningOnSameThread(clientboundContainerAckPacket, this, this.minecraft);
  AbstractContainerMenu abstractContainerMenu = null;
  LocalPlayer localPlayer = this.minecraft.player;
  if (clientboundContainerAckPacket.getContainerId() == 0) {
      abstractContainerMenu = localPlayer.inventoryMenu;
  } else if (clientboundContainerAckPacket.getContainerId() == localPlayer.containerMenu.containerId) {
      abstractContainerMenu = localPlayer.containerMenu;
  }
  if (abstractContainerMenu != null && !clientboundContainerAckPacket.isAccepted()) {
      this.send(new ServerboundContainerAckPacket(clientboundContainerAckPacket.getContainerId(), clientboundContainerAckPacket.getUid(), true));
  }
}